### PR TITLE
Add details on the "Discover relation" entry

### PR DESCRIPTION
**Pull request by DelazJ**
Also move the multiple pairing suggestion to a note to avoid confusion since we stated earlier that a single pair is to be used with the example dataset

### CODE REVIEW

File: docs/user_manual/working_with_vector/attribute_table.rst [hunk 1]
```diff
@@ -680,25 +680,30 @@ for editing and QGIS takes care of the setup. It works with different
...
-between the layers. This is done in :menuselection:`Project --> Project Properties...`.
-Open the :guilabel:`Relations` tab and click on :guilabel:`Add Relation`.
+between the layers. This is done in :menuselection:`Project --> Properties...`.
+Open the :guilabel:`Relations` tab and click on |signPlus| :guilabel:`Add Relation`.
 
 * **Name** is going to be used as a title. It should be a human readable string,
   describing, what the relation is used for. We will just call say **airport_relation**
   in this case.
 * **Referenced Layer (Parent)** also considered as parent layer, is the one with
   the primary key, pointed to, so here it is the ``regions`` layer. You can define
```
Comment (m-kuhn) [inline docs/user_manual/working_with_vector/attribute_table.rst:694]:
```suggestion
  the primary key, pointed to, so here it is the ``regions`` layer. You need to define
```

File: docs/user_manual/working_with_vector/attribute_table.rst [hunk 2]
```diff
@@ -680,25 +680,30 @@ for editing and QGIS takes care of the setup. It works with different
...
 * **Referencing Layer (Child)** also considered as child layer, is the one with
   the foreign key field on it. In our case, this is the ``airports`` layer. For
   this layer you need to add a referencing field which points to the other
-  layer, so this is ``fk_region``. When using multiple field relations you can
-  add another referencing field by using the |signPlus| button.
+  layer, so this is ``fk_region``.
+
+  .. note:: Sometimes, you need more than a single field to uniquely identify
+   features in a layer. Creating a relation with such a layer requires
+   a **composite key**, ie more than a single pair of referencing/referenced
```
Comment (m-kuhn) [inline docs/user_manual/working_with_vector/attribute_table.rst:703]:
```suggestion
   a **composite key**, ie more than a single pair of matching
```

File: docs/user_manual/working_with_vector/attribute_table.rst [hunk 3]
```diff
@@ -714,7 +719,12 @@ Open the :guilabel:`Relations` tab and click on :guilabel:`Add Relation`.
 .. figure:: img/relations2.png
    :align: center
 
-   Relation Manager
+   Adding a relation between regions and airports layers
+
+From the :guilabel:`Relations` tab, you can also press the |signPlus|
+:guilabel:`Discover Relation` button to fetch the relations available from
+the providers of the current layers. This is possible for layers stored in
+data providers like PostgreSQL or SpatiaLite.
```
Comment (m-kuhn) [inline docs/user_manual/working_with_vector/attribute_table.rst:727]: Doesn't it also work with GeoPackage?

File: docs/user_manual/working_with_vector/attribute_table.rst [hunk 4]
```diff
@@ -714,7 +719,12 @@ Open the :guilabel:`Relations` tab and click on :guilabel:`Add Relation`.
 .. figure:: img/relations2.png
    :align: center
 
-   Relation Manager
+   Adding a relation between regions and airports layers
+
+From the :guilabel:`Relations` tab, you can also press the |signPlus|
+:guilabel:`Discover Relation` button to fetch the relations available from
+the providers of the current layers. This is possible for layers stored in
```
Comment (m-kuhn) [inline docs/user_manual/working_with_vector/attribute_table.rst:726]: What are "current layers"?

Comment (DelazJ) [reply docs/user_manual/working_with_vector/attribute_table.rst:726]:
```suggestion
the providers of the loaded layers. This is possible for layers stored in
```
?

Comment (m-kuhn) [reply docs/user_manual/working_with_vector/attribute_table.rst:726]: :+1:

**Comment by ghtmtt** [comment]
nice addition , thanks @DelazJ!

File: docs/user_manual/working_with_vector/attribute_table.rst (hunk 3 above)
Comment (DelazJ) [reply docs/user_manual/working_with_vector/attribute_table.rst:727]: Good question, and the reason why I wanted a dev around ;). I found a discover_relation in both postgresql, spatialite (and ogr - I don't know which format) classes: https://github.com/qgis/QGIS/search?q=discoverrelations&type=Code. Moreover, in https://docs.qgis.org/3.10/en/docs/user_manual/working_with_vector/attribute_table.html#introducing-many-to-many-n-m-relations it's mentioned that the "discover relation" can be used only for PG and not gpkg (obsolete information or still true? I don't know and can't reasonably check myself)

Comment (m-kuhn) [reply docs/user_manual/working_with_vector/attribute_table.rst:727]: Where did you find it in ogr? That would hint to gpkg but I didn't see it in the search results (which would be a hint it's not supported)

File: docs/user_manual/working_with_vector/attribute_table.rst [hunk 5]
```diff
@@ -680,25 +680,30 @@ for editing and QGIS takes care of the setup. It works with different
...
   the foreign key field on it. In our case, this is the ``airports`` layer. For
   this layer you need to add a referencing field which points to the other
-  layer, so this is ``fk_region``. When using multiple field relations you can
-  add another referencing field by using the |signPlus| button.
+  layer, so this is ``fk_region``.
+
+  .. note:: Sometimes, you need more than a single field to uniquely identify
+   features in a layer. Creating a relation with such a layer requires
+   a **composite key**, ie more than a single pair of referencing/referenced
+   fields. Use the |signPlus| :sup:`Add fields to pair` button
```
Comment (DelazJ) [inline docs/user_manual/working_with_vector/attribute_table.rst:704]:
```suggestion
   fields. Use the |signPlus| :sup:`Add new field pair as part of a composite
```

File: docs/user_manual/working_with_vector/attribute_table.rst [hunk 6]
```diff
@@ -680,25 +680,30 @@ for editing and QGIS takes care of the setup. It works with different
...
+   to add as many pairs as necessary.
```
Comment (DelazJ) [inline docs/user_manual/working_with_vector/attribute_table.rst:705]:
```suggestion
   foreign key` button to add as many pairs as necessary.
```

File: docs/user_manual/working_with_vector/attribute_table.rst (hunk 3 above)
Comment (DelazJ) [reply docs/user_manual/working_with_vector/attribute_table.rst:727]: Oups. Sorry. I misread vectorprovider as ogr.

Comment (m-kuhn) [reply docs/user_manual/working_with_vector/attribute_table.rst:727]: In this case, let's leave it as is. Possibly open a feature request for gopackage support?

Comment (DelazJ) [reply docs/user_manual/working_with_vector/attribute_table.rst:727]: Feature request opened at https://github.com/qgis/QGIS/issues/38013

**Pull request merged**